doc/specs/OOBE-Dialog.md

```diff
@@ -0,0 +1,169 @@
+
+### 2.2. Expected Impact: Customer, and Technology Outcomes
+
+- **High Reliability:** Less than 0.1% crash rate.
+- **Increased Activation:** 30% or more increase in usage in all PowerToys features.
+- **High User Retention:** 25% or more active PowerToys users after 28 days.
+- **High OOBE Completion Rate:** 50%  or more of New PowerToys users reach the last page of the OOBE prompt.
+
+## 3. Requirements
+
+The base of this OOBE dialog was attributed by [Niel’s mock-up](https://github.com/microsoft/PowerToys/issues/1285).
+
+### 3.1. Functional Requirements
+
+#### 3.1.1. Functional Requirements Overview
+
+|No. | Requirement | Pri |
+| - | - | - |
+|1 | The OOBE Dialog window should launch immediately after installation. | P0 |
+|2 | The OOBE Dialog should consist of a collection of pages that describe the available PowerToys. | P0 |
+|3 | On launch, present the Welcome/Landing page in the OOBE Dialog window based on the template described in [Figure 5.1.1.](#511-oobe-dialog-layout) | P1 |
+|4 | Traverse the various OOBE Dialog pages via an list view options on the left-hand pane. | P0 |
+|5 | Alternatively traverse the various OOBE Dialog pages via an on-screen next/previous buttons on each page. | P1 |
+|6 | At any time, the user should be able to exit the OOBE dialog by closing the window. | P0 |
+|7 | Alternatively, the user should be able to exit the OOBE dialog by pressing the ESC key. | P1 |
+|8 | When the OOBE dialog is exited, save all the options that were set by the user. | P0 |
+|9 | In the PowerToys settings “General” menu, add a button to re-launch the OOBE dialog under the “Run at startup” option. | P0 |
+|10 | The OOBE dialog window should be able to be enlarged to accommodate accessibility needs. | P0 |
+|11 | The OOBE dialog should have a preset minimum window size to ensure all content can be viable with the need to scroll the page. | P1 |
+
+#### 3.1.2. Page Layout
+
+| No. | Requirement | Pri |
+| - | - | - |
+|1 | Each page of the OOBE dialog should focus on the details and settings of a single PowerToy. | P0 |
+|2 | A page should include textual information that gives an overview of its PowerToy's basic functions. | P0 |
+|3 | A page should include a hero image/GIF that provides a visual representation of its PowerToy's core functionality. | P0 |
+|4 | A page should include a link to that PowerToys overview page on Microsoft Docs. | P0 |
+|5 | For applicable PowerToys, include a &quot;Launch&quot; button on the page to allow users to instantiate the PowerToy directly from the OOBE dialog window. | P1 |
+|6 | Include a &quot;Settings&quot; button that deep links to the PowerToy's settings page. | P1 |
+|7 | A page should include a short-list of relevant settings options that the user can select between to customize their experience. | P1 |
```

Comment:
This need to be spec'ed out in details, because it can lead to inconsistencies, for example:
 - user changes an option for this module, but given 3.1.1.8, that settings should only be saved when OOBE exists
 - user stays on the same page and not clicks the link that open the Settings page for that module that will show a different value for that option
 - user now makes changes directly in the module's Settings page and those are saved instantly
 - the user exists OOBE that is going to overwrite the Settings saved previously

Reply:
@enricogior Yeah makes sense. I agreed with you earlier that the settings should be automatically saved as they are changed, regardless of whether that's in OOBE or the settings page. Is this enough clarity, or can you think of any more inconsistencies that require more details to be spec'd out?